enhancements/olm/safe-cluster-upgrades.md

```diff
@@ -0,0 +1,270 @@
+
+Examples can be found below:
+
+| OpenShift Cluster Version   | CSV's Max OpenShift Version | Is Upgradeable?|
+| --------------------------- | --------------------------- | -------------- |
+| 4.6.z                       | 4.8.0                       | True           |
+| 4.7.z                       | 4.8.0                       | True           |
+| 4.8.z                       | 4.8.0                       | False          |
+
+As this is an opt-in feature, there are instances where OLM will be unable to determine if the upgrade is safe:
+
+- The CSV is missing the `operators.coreos.com/maxOpenShiftVersion` annotation
+- The value associated with the `operators.coreos.com/maxOpenShiftVersion` is not a valid Semantic Version (Valid Example: x.y.z).
+
+OLM will report the following upgradeable conditions based on the collection of CSV present on cluster:
+
+| Key                  | Meaning                                                                                            |
+| ---------------------| -------------------------------------------------------------------------------------------------- |
+| Upgradeable    CSV   | A CSV with a valid upgradeable annotation that will run on the next Minor version of OpenShift     |
+| Non-Upgradeable  CSV | A CSV with a valid upgradeable annotation that will not run on the next Minor version of OpenShift |
+| Undeterminable CSV   | A CSV that does not include a valid upgradeable annotation                                         |
+
+| Upgradeable CSV Present | Indeterminate CSV Present | Non-Upgradeable CSV Present | Status          | Message                                                                                            |
+| ----------------------- | ------------------------- | --------------------------- |---------------- | -------------------------------------------------------------------------------------------------- |
+| Yes                     | No                        | No                          | Upgradeable     | Ready for upgrade                                                                                  |
+| Don't Care              | Yes                       | No                          | Upgradeable     | The following operators may not run on the next OpenShift Version: namespace/foo, namespace/bar"   |
+| Don't Care              | Don't Care                | Yes                         | Not Upgradeable | The following operators will not run on the next OpenShift Version: namespace/foo, namespace/bar"  |
+
+#### Pros
+
+- This approach provides allows Operator Authors to specify the exact versions of OpenShift that their operator supports.
+
+#### Cons
+
+- This makes upgrade safety opt-in for each Operator, meaning that OLM cannot guarantee upgrade safety unless each installed operator specifies a maxOpenShiftVersion.
```

Comment:
And even then, these are declarations about compatibility with a version you are not yet running.  Folks can get declarations wrong.  Even if we test operator A on OpenShift B, and it works under testing, that does not guarantee it will work on B for every customer.  Bugs happen.  I'd soften "guarantee" -> "calculate" or something.